### PR TITLE
All new profiles for the Camunda Platform Helm chart

### PR DESCRIPTION
**Pull request by falko**
The content of this repository is pretty outdated as points to the old Zeebe Helm charts. I suggest to switch out the content for new profiles that we developed together with customers.

Also after merging this PR, the repository should be renamed to `camunda-8-helm-profiles`.

### CODE REVIEW

File: include/camunda.mk [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.PHONY: camunda
```
Comment (aabouzaid) [inline include/camunda.mk:1]: I'd suggest to rename the dir to `make` for clarity.
If you plan to add more includes, then it should be `include/make`.

**Comment by aabouzaid** [comment]
It would be a bit hard to review this in detail, but in general, I'd suggest making a structure in the repo dirs to make it easy to navigate and work with.

**Review by Zelldon — COMMENTED** [review]
Not sure what I look for. What is now the difference to https://github.com/camunda-community-hub/camunda8-greenfield-installation can we merge both?

File: include/camunda.mk [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+.PHONY: template
+template:
+	helm template $(release) $(chart) -f $(chartValues) --skip-crds --output-dir .
+	@echo "To apply the templates use: kubectl apply -f camunda-platform/templates/ -n $(namespace)"
+
+.PHONY: update
+update:
+	OPERATE_SECRET=$$(kubectl get secret --namespace $(namespace) "camunda-operate-identity-secret" -o jsonpath="{.data.operate-secret}" | base64 --decode); \
+	TASKLIST_SECRET=$$(kubectl get secret --namespace $(namespace) "camunda-tasklist-identity-secret" -o jsonpath="{.data.tasklist-secret}" | base64 --decode); \
+	OPTIMIZE_SECRET=$$(kubectl get secret --namespace $(namespace) "camunda-optimize-identity-secret" -o jsonpath="{.data.optimize-secret}" | base64 --decode); \
```
[the remark this answers is not in the document]
Comment (falko) [reply include/camunda.mk:23]: Do you happen to know the variables they use?

File: include/camunda.mk [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+camunda: namespace
+	helm repo add camunda https://helm.camunda.io
+	helm repo update camunda
+	helm search repo $(chart)
+	helm install --namespace $(namespace) $(release) $(chart) -f $(chartValues) --skip-crds
+
+.PHONY: namespace
+namespace:
+	-kubectl create namespace $(namespace)
+	-kubens $(namespace)
```
[the remark this answers is not in the document]
Comment (falko) [reply include/camunda.mk:11]: Does `kubectl` have a native feature to remember the active namespace or do you mean setting the namespace for every command, which we already do.

File: include/camunda.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zelldon) [reply include/camunda.mk:23]: Its in the docs :grin: https://docs.camunda.io/docs/self-managed/platform-deployment/kubernetes-helm/#secrets-extraction

**Comment by falko** [comment]
> Not sure what I look for. What is now the difference to https://github.com/camunda-community-hub/camunda8-greenfield-installation can we merge both?

Please see my last two comments in: https://github.com/camunda/camunda-platform-helm/pull/373#issuecomment-1216434646

**Comment by falko** [comment]
Another way to decide on this pull request is to ask: I anyone even feeling responsible as a maintainer for this repo, maybe @Zelldon? Or is this as abandoned as its commit history indicates and you're happy for @upgradingdave and me giving it a new try?

**Comment by Zelldon** [comment]
@falko in my opinion you could just go ahead and do what you want with the repo :sweat_smile: I would prefer one repo instead of the two (speaking of the profiles and the greenfield repo).

**Comment by falko** [comment]
> I would prefer one repo instead of the two (speaking of the profiles and the greenfield repo).

Let's discuss this in a [separate issue](https://github.com/camunda-community-hub/camunda8-greenfield-installation/issues/3) so that we can make progress with this PR independently.

File: include/camunda.mk (hunk 1 above)
Comment (falko) [reply include/camunda.mk:1]: I tried to address you feedback with a README file in the folder: https://github.com/falko/camunda-8-helm-profiles/tree/master/include#readme

File: include/camunda.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (falko) [reply include/camunda.mk:11]: @upgradingdave could you please replace `kubens` with the above command and remove it from the prerequisites?

File: include/camunda.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (falko) [reply include/camunda.mk:23]: @upgradingdave could you please add the missing secrets for Keycloak and Postgres as documented in the link above?

**Comment by Zelldon** [comment]
@falko do we still need the meeting tomorrow? :thinking:

**Comment by falko** [comment]
Let's have a chat about the KIND and OpenShift profiles and the proposed repo merge.

**Comment by Zelldon** [comment]
I guess then @npepinpe should be invited as well :man_shrugging:

File: include/camunda.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (upgradingdave) [reply include/camunda.mk:11]: I've made this change locally and testing it now. Thanks for the tip, @Zelldon! I'll push this change as soon as I verify I didn't break anything.

[the remark this answers is not in the document]
Comment (upgradingdave) [reply include/camunda.mk]: I made the change and tested in GKE. Seems to work well. With this change, after running `make camunda`, any subsequent `kubectl` calls are bound to the `camunda` namespace by default.

File: include/camunda.mk (hunk 2 above)
[the remark this answers is not in the document]
Comment (upgradingdave) [reply include/camunda.mk]: Added the missing secrets for keycloak and postgres. Tested in GKE and was able to access the apps so I think it's good, although I'll test more as I have time to make sure this didn't break anything.

**Comment by falko** [comment]
Merging now since all change requests have been resolved.

**Pull request merged**